File: src/main/java/com/antero/liemipeli/domain/opiskelu/Opettaja.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.antero.liemipeli.domain.opiskelu;

import com.antero.liemipeli.domain.ElavaOlio;
import com.antero.liemipeli.domain.opiskelu.Kurssi;
import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

/**
 *
 * @author Antero Oikkonen
 */
public class Opettaja extends ElavaOlio{
    //  opettaja voi olla pätevä opettamaan erilaisia opettavia aineita
    private List <Kurssi> opAineet;
    private String oppiarvo;

    public List<Kurssi> getOpAineet() {
        return opAineet;
    }

    public void setOpAineet(List<Kurssi> opAineet) {
        this.opAineet = opAineet;
    }


    public String getOppiarvo() {
        return oppiarvo;
    }

    public void setOppiarvo(String oppiarvo) {
        this.oppiarvo = oppiarvo;
    }

    public void addKurssi(Kurssi kurssi) {
        if(opAineet==null)
            opAineet=new ArrayList<Kurssi>();
        
        opAineet.add(kurssi);
    }

    @Override
    public String toString() {
        return getNimi()+", \n"+getOppiarvo()+",\n"+getOpAineet();
    }
}

File: liemipeli-kotlin/src/main/kotlin/com/antero/liemipeli/opiskelu/Opettaja.kt
package com.antero.liemipeli.opiskelu

import com.antero.liemipeli.domain.elaimet.ElavaOlio


/**
 *
 * @author Antero Oikkonen
 */
class Opettaja(nimi: String) : ElavaOlio {

    val eleva: Boolean

    override fun isElava(): Boolean {
       return this.eleva
    }

    init {
        eleva = true
    }


    //  opettaja voi olla pätevä opettamaan erilaisia opettavia aineita
    private var opAineet: MutableList<Kurssi>? = null
    var oppiarvo: String? = null
    override val nimi = nimi

    fun getOpAineet(): List<Kurssi>? {
        return opAineet
    }

    fun setOpAineet(opAineet: MutableList<Kurssi>) {
        this.opAineet = opAineet
    }

    fun addKurssi(kurssi: Kurssi) {
        if (opAineet == null)
            opAineet = ArrayList()

        opAineet!!.add(kurssi)
    }

    override fun toString(): String {
        return "\nOpettajan nimi: ${nimi}\nOppiarvo: ${oppiarvo}\nOppiaineet: ${opAineet}"
    }
}
